penman/tree.py

```diff
@@ -180,15 +182,32 @@ def _map_vars(
 
     newbranches: List[Branch] = []
     for role, tgt in branches:
-        if not is_atomic(tgt):
+        if is_tgt_node(tgt):
             tgt = _map_vars(tgt, varmap)
-        elif role != '/' and tgt in varmap:
+        # MyPy forgets that (Node ∨ Sym) ^ ¬Node → Sym
+        elif is_tgt_symbol(tgt) and role != '/' and tgt in varmap:
             tgt = varmap[tgt]
         newbranches.append((role, tgt))
 
     return (varmap[var], newbranches)
 
 
+def is_tgt_node(target: Symbol | Node) -> TypeGuard[Node]:
+    """
+    Inverse of :func:`is_atomic`, only for Symbol | Node from branches.
+    Automatically narrows the type to Node for better type inference
+    """
+    return not is_atomic(target)
+
+
+def is_tgt_symbol(target: Symbol | Node) -> TypeGuard[Symbol]:
+    """
+    Same as :func:`is_atomic`, only for Symbol | Node from branches.
+    Automatically narrows the type to Symbol for better type inference
+    """
+    return is_atomic(target)
```

Comment:
There are a few issues here:
1. It's not great that these don't really do anything different from `is_atomic()` except for type checking. I think having them as part of the public API would confuse users.
2. If they were to stay, I'd prefer to use non-abbreviated names in public API functions: `is_target_symbol()` or maybe `is_symbol()`.
3. `TypeGuard` is added in Python 3.10, but Penman currently supports down to Python 3.8.